dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/DspaceObjectMetadataCopyOperation.java

```diff
@@ -0,0 +1,102 @@
+ * </code>
+ *
+ * @author Maria Verdonck (Atmire) on 18/11/2019
+ */
+@Component
+public class DspaceObjectMetadataCopyOperation<R extends DSpaceObject> extends PatchOperation<R> {
+
+    @Autowired
+    DspaceObjectMetadataPatchUtils metadataPatchUtils;
+
+    @Override
+    public R perform(Context context, R resource, Operation operation) throws SQLException {
+        DSpaceObjectService dsoService = ContentServiceFactory.getInstance().getDSpaceObjectService(resource);
+        MetadataField metadataField = metadataPatchUtils.getMetadataField(context, operation);
+        String[] partsFromCopy = ((CopyOperation) operation).getFrom().split("/");
+        String indexToCopyFrom = (partsFromCopy.length > 3) ? partsFromCopy[3] : null;
+
+        copy(context, resource, dsoService, metadataField, indexToCopyFrom);
+        return resource;
+    }
+
+    /**
+     * Copies metadata of the dso from indexFrom to new index at end of md
+     *
+     * @param context         context patch is being performed in
+     * @param dso             dso being patched
+     * @param dsoService      service doing the patch in db
+     * @param metadataField   md field being patched
+     * @param indexToCopyFrom index we're copying metadata from
+     */
+    private void copy(Context context, DSpaceObject dso, DSpaceObjectService dsoService, MetadataField metadataField,
+                      String indexToCopyFrom) {
+        metadataPatchUtils.checkMetadataFieldNotNull(metadataField);
+        List<MetadataValue> metadataValues = dsoService.getMetadata(dso, metadataField.getMetadataSchema().getName(),
+                metadataField.getElement(), metadataField.getQualifier(), Item.ANY);
+        try {
+            int indexToCopyFromInt = Integer.parseInt(indexToCopyFrom);
+            if (indexToCopyFromInt >= 0 && metadataValues.size() > indexToCopyFromInt
+                    && metadataValues.get(indexToCopyFromInt) != null) {
+                MetadataValue metadataValueToCopy = metadataValues.get(indexToCopyFromInt);
+                MetadataValueRest metadataValueRestToCopy
+                        = metadataPatchUtils.convertMdValueToRest(metadataValueToCopy);
+                // Add metadata value to end of md list
+                dsoService.addAndShiftRightMetadata(context, dso, metadataField.getMetadataSchema().getName(),
+                        metadataField.getElement(), metadataField.getQualifier(), metadataValueRestToCopy.getLanguage(),
+                        metadataValueRestToCopy.getValue(), metadataValueRestToCopy.getAuthority(),
+                        metadataValueRestToCopy.getConfidence(), -1);
+            } else {
+                throw new UnprocessableEntityException("There is no metadata of this type at that index");
+            }
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("This index (" + indexToCopyFrom + ") is not valid nr", e);
```

Comment:
Could we change "not valid nr" to "not valid number"?  The meaning of "nr" was not immediately clear to me.